io/zenoh-transport/src/unicast/link.rs

```diff
@@ -391,3 +470,63 @@ async fn rx_task(
         rx_task_dgram(link, transport, lease, signal, rx_buffer_size).await
     }
 }
+
+#[cfg(feature = "transport_compression")]
+fn tx_compressed(
+    compression_enabled: bool,
+    bytes: &[u8],
+    compression_aux_buff: &mut Box<[u8]>,
+    link: &LinkUnicast,
+) -> ZResult<Vec<u8>> {
+    let mut buff = vec![];
+    if compression_enabled {
+        let zenoh_compress = ZenohCompress::default();
+        let compression_size = zenoh_compress
+            .write(&mut buff.writer(), (&bytes, compression_aux_buff))
+            .map_err(|e| zerror!("Compression error: {:?}", e))?;
+        let mut compressed_batch: Vec<u8> = vec![];
+        if link.is_streamed() {
+            let compression_size_u16: u16 = compression_size.try_into().map_err(|e| {
+                zerror!(
+                    "Compression error: unable to convert compression size into u16: {}",
+                    e
+                )
+            })?;
+            // including is compressed byte
+            let batch_size_le = (compression_size_u16 + 1).to_le_bytes();
+            compressed_batch.append(&mut batch_size_le.to_vec());
+            compressed_batch.push(true as u8);
+            compressed_batch.append(&mut buff);
+        } else {
+            compressed_batch.push(true as u8);
+            compressed_batch.append(&mut buff);
+        }
+        buff = compressed_batch;
+    } else {
+        // Add compression byte as false
+        // Increment the batch size number from the 2 initial bytes
+        buff = bytes.to_vec();
+        buff.insert(2, false as u8);
+        let (batch_size, batch_payload) = buff.split_at_mut(2);
+        let batch_size = u16::from_le_bytes(batch_size.try_into().unwrap()) + 1;
+        buff = [batch_size.to_le_bytes().to_vec(), batch_payload.to_vec()].concat();
+    }
+    Ok(buff)
+}
+
+#[cfg(feature = "transport_compression")]
+fn rx_decompress(
+    zenoh_compress: &ZenohCompress,
+    compression: &[u8],
+    aux_buffer: &mut [u8],
+) -> ZResult<usize> {
+    Ok(zenoh_compress
+        .read((&compression, aux_buffer))
+        .map_err(|e| {
+            zerror!(
+                "Decompression error {:?}. Unable to decompress batch. {:?}",
+                e,
+                compression.to_vec()
```

Comment:
There is no need to call to_vec() here. Moreover, if you want to print out the batch, I'd rather use log::trace! rather then putting in as string in the error message.